src/input_context/input_action.rs

```diff
@@ -254,27 +251,32 @@ pub enum ActionState {
 /// derive to reduce boilerplate:
 ///
 /// ```
+/// # use bevy::prelude::*;
 /// # use bevy_enhanced_input::prelude::*;
 /// #[derive(Debug, InputAction)]
-/// #[input_action(dim = Axis2D)]
+/// #[input_action(output = Vec2)]
 /// struct Move;
 /// ```
 ///
 /// Optionally you can pass `consume_input` and/or `accumulation`:
 ///
 /// ```
+/// # use bevy::prelude::*;
 /// # use bevy_enhanced_input::prelude::*;
 /// #[derive(Debug, InputAction)]
-/// #[input_action(dim = Axis2D, accumulation = Cumulative, consume_input = false)]
+/// #[input_action(output = Vec2, accumulation = Cumulative, consume_input = false)]
 /// struct Move;
 /// ```
 pub trait InputAction: Debug + Send + Sync + 'static {
-    /// Discriminant for [`ActionValue`] that will be used for this action.
+    /// What type of value this action will output.
+    ///
+    /// - Use [`bool`] for button-like actions (e.g., `Jump`).
+    /// - Use [`f32`] for single-axis actions (e.g., `Zoom`).
+    /// - For multi-axis actions, like `Move`, use [`Vec2`] or [`Vec3`].
     ///
-    /// Use [`ActionValueDim::Bool`] for button-like actions (e.g., `Jump`).
-    /// Use [`ActionValueDim::Axis1D`] for single-axis actions (e.g., `Zoom`).
-    /// For multi-axis actions, like `Move`, use [`ActionValueDim::Axis2D`] or [`ActionValueDim::Axis3D`].
-    const DIM: ActionValueDim;
+    /// The type here will determine the type of the `value` field on events
+    /// e.g. [`Fired::value`], [`Canceled::value`].
+    type Output: ActionValueOutput;
```

Comment:
Maybe just require `Into<ActionValue>`? It's already implemented for `bool`, `f32` and Vec's.

Reply:
`ActionValueOutput` is used for multiple reasons:
- it can be sealed, so only `bool`, `f32`, `Vec2` and `Vec3` implement it, since those are the only `ActionValue` variants
- `const DIM` is needed to get the `ActionValueDim` of the type
- `convert_from(ActionValue)`  but this can be replaced with `Into<ActionValue>`

Reply:
Agree, let's keep a separate trait.
But do we need to seal it? If someone wants to use another type, is there a possible footgun? :thinking: I would remove sealing and for conversion just use the `Into` impl to avoid duplication (just add the requirement for the trait).

Reply:
I suppose someone *could* use another type, and impl `ActionValueOutput` manually. But I don't see any real reason to do that, and this would also mean that `ActionValueOutput` is part of the public API and would have to follow semver. So we couldn't modify its internals without it counting as a breaking change.

I see this trait as more of an implementation detail, so I would keep it sealed. Maybe even move the public const and fn into the `sealed` trait.

Reply:
> But I don't see any real reason to do that

Me too. But I prefer to seal types only if using them might cause a footgun. Sealed types just don't play nice with the docs. Since it's a very simple trait, I would allow user to see it, just don't add it to the prelude.